lib/smart_proxy_remote_execution_ssh/api.rb

```diff
@@ -1,9 +1,133 @@
+require 'net/ssh'
+
 module Proxy::RemoteExecution
   module Ssh
     class Api < ::Sinatra::Base
+      include Sinatra::Authorization::Helpers
+
       get "/pubkey" do
         File.read(Ssh.public_key_file)
       end
+
+      post "/session" do
+        do_authorize
```

Comment:
@mvollmer I've noticed this endpoint was not properly authorized (I'm glad we've noticed this before releasing: saved use some troubles:).

It depends on https://github.com/theforeman/smart-proxy/pull/639 getting in first, so that we can actually use the authorization per-endpoint.

Reply:
I get 
```
NameError: undefined local variable or method `do_authorize' for #<Proxy::RemoteExecution::Ssh::Api:0x007ffa4c3c70f0>
/usr/share/gems/gems/smart_proxy_remote_execution_ssh-0.2.0/lib/smart_proxy_remote_execution_ssh/api.rb:34:in `block in <class:Api>'
```

Indeed, I can't find `do_authorize` in https://github.com/theforeman/smart-proxy/pull/639

Reply:
It has been updated as part of the review: now updated here as well